dotnet/src/SemanticKernel.Abstractions/Orchestration/FunctionResult.cs

```diff
@@ -80,4 +80,20 @@ public FunctionResult(string functionName, string pluginName, SKContext context,
 
         throw new InvalidCastException($"Cannot cast {this.Value.GetType()} to {typeof(T)}");
     }
+
+    /// <summary>
+    /// Get typed value from metadata.
+    /// </summary>
+    public bool TryGetMetadataValue<T>(string key, out T value)
```

Comment:
Given that this is a public method, some tests for this could help prevent regressions.